### PR TITLE
Protobuf Formatting and Style Guide

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Add protobuf support to formatting and add a section on style guide. Formatting is the same as cpp. I've decided that we can format SSL protos since we have to manually port them over when they change anyways.

### Testing Done
CI

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — APPROVED** [review]
LGTM :fire: nice work!

**Review by MathewMacDougall — APPROVED** [review]
LGTM :fire:

File: src/software/proto/messages_robocup_ssl_geometry.proto [hunk 1]
```diff
@@ -1,71 +1,77 @@
...
-  required Vector2f center = 2;
-  // Radius of the arc.
-  required float radius = 3;
-  // Start angle in counter-clockwise order.
-  required float a1 = 4;
-  // End angle in counter-clockwise order.
-  required float a2 = 5;
-  // Thickness of the arc.
-  required float thickness = 6;
+message SSL_FieldCicularArc
```
Comment (kelvinkoon) [inline src/software/proto/messages_robocup_ssl_geometry.proto:27]: "Cicular" seems to be a misspelling. Even though it was previously misspelled as well, should it be fixed?

**Review by kelvinkoon — APPROVED** [review]
LGTM besides 1 comment :+1:

File: src/software/proto/messages_robocup_ssl_geometry.proto (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/proto/messages_robocup_ssl_geometry.proto:27]: :eyes: !

This is from the SSL, so we should double check this still exists on their master branch, and open a PR to fix it if necessary. (This if from SSL Vision)

Comment (jonathanlew) [reply src/software/proto/messages_robocup_ssl_geometry.proto:27]: yes it exists on their master :grimacing: yikes

Comment (kelvinkoon) [reply src/software/proto/messages_robocup_ssl_geometry.proto:27]: Then we should keep it as `SSL_FieldCicularArc` until they fix it on their master branch so LGTM :100:

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Comment by jonathanlew** [comment]
Created an issue: https://github.com/RoboCup-SSL/ssl-vision/issues/170

File: src/software/proto/messages_robocup_ssl_geometry.proto (hunk 1 above)
Comment (jonathanlew) [reply src/software/proto/messages_robocup_ssl_geometry.proto:27]: Created an issue: RoboCup-SSL/ssl-vision#170

**Pull request merged**